es5-shim.js

```diff
@@ -328,6 +329,73 @@ var strSplit = call.bind(StringPrototype.split);
 var strIndexOf = call.bind(StringPrototype.indexOf);
 var push = call.bind(array_push);
 
+// ES-5 15.3.4.4
+// http://es5.github.io/#x15.3.4.4
+var hasToStringTagBasicBug = toStr() !== '[object Undefined]' || toStr(null) !== '[object Null]';
+var hasToStringTagLegacyArguments = toStr(arguments) !== '[object Arguments]';
+var hasToStringTagInconsistency = hasToStringTagBasicBug || hasToStringTagLegacyArguments;
+
+// Others to check:
+// PhantomJS 1.9 which returns 'function' for `NodeList` instances.
+// Older ES3 native functions like `alert`, there are some fixes available but
+// are they worth it?
+
+if (hasToStringTagLegacyArguments) {
+  var isDuckTypeArguments = function (value) {
+    return value != null &&
+        typeof value === 'object' &&
+        owns(value, 'length') &&
+        typeof value.length === 'number' &&
+        value.length >= 0 &&
+        owns(value, 'callee') &&
+        isCallable(value.callee) &&
+        !owns(value, 'arguments');
+
+  };
+}
+
+if (hasToStringTagInconsistency) {
+    var baseSlice = function (array, start, end) {
+        var begin = start == null ? 0 : start >> 0;
+        var length = array.length;
+        if (begin < 0) {
+            begin = -begin > length ? 0 : (length + begin);
+        }
+        var finish = (typeof end === 'undefined' || end > length) ? length : end >> 0;
+        if (finish < 0) {
+            finish += length;
+        }
+        length = begin > finish ? 0 : (finish - begin) >>> 0;
+        begin = begin >>> 0;
+        var index = -1;
+        var result = Array(length);
+        while (++index < length) {
+            result[index] = array[index + begin];
+        }
+        return result;
+    };
+
+    defineProperties(FunctionPrototype, {
+      call: function () {
+          var thisArg = arguments[0];
+          if (this === to_string) {
+              if (hasToStringTagBasicBug) {
```

Comment:
We only want `call` to give us a `[[Class]]` if the context is `Object#toString`, otherwise we want it to go about its normal business of call whatever the function `this` is.